src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -1,47 +1,54 @@
 package hu.bme.mit.spaceship;
 
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
 import java.util.Random;
 
 /**
-* Class storing and managing the torpedoes of a ship
-*
-* (Deliberately contains bugs.)
-*/
+ * Class storing and managing the torpedoes of a ship
+ *
+ * (Deliberately contains bugs.)
+ */
 public class TorpedoStore {
 
   // rate of failing to fire torpedos [0.0, 1.0]
-  private double FAILURE_RATE = 0.0; //NOSONAR
+  private double FAILURE_RATE = 0.0; // NOSONAR
 
   private int torpedoCount = 0;
+  private Random rand;
 
-  public TorpedoStore(int numberOfTorpedos){
+  public TorpedoStore(final int numberOfTorpedos) {
     this.torpedoCount = numberOfTorpedos;
+    try {
+      this.rand = SecureRandom.getInstanceStrong();
+    } catch (NoSuchAlgorithmException e) {
+      e.printStackTrace();
+    }
```

Comment:
Don't use printStackTrace() as it is a potential security vulnerability.

Either use new Random() instead of SecureRandom.getInstanceStrong() so you don't have to catch any exceptions
OR
use a logger

Reply:
It's a fantastic idea. Well done.